src/utils/use-previous.ts

```diff
@@ -0,0 +1,9 @@
+import {useEffect, useRef} from 'react';
+
+export const usePrevious = <T>(value: T): T | undefined => {
+  const ref = useRef<T>();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
```

Comment:
I copied this from a SO post, but when looking at it I wondered why `value` wasn't a part of the dependency array in `useEffect`. When looking at other sources for similar custom hook, I see they all have `value` in dependency array.
https://usehooks.com/usePrevious/
https://blog.logrocket.com/accessing-previous-props-state-react-hooks/